src/api/functionalHelpers.js

```diff
@@ -0,0 +1,28 @@
+import curry from 'lodash/curry';
+
+/**
+ * given a value an defautl value, act as identity function except if value is undefined
+ * returning default value instead.
+ * @param {a} defaultValue
+ * @param {b} value
+ * @return {a | b}
+ */
+export const withDefault = curry((defaultValue, value) => {
+	if (value === undefined) {
```

Comment:
what about `null` value ?

Reply:
yep going to add coverage on this one, also present in the streams app